src/coreclr/jit/valuenum.cpp

```diff
@@ -8407,14 +8423,29 @@ void Compiler::fgValueNumberTree(GenTree* tree)
             case GT_LCL_FLD:
             {
                 GenTreeLclFld* lclFld = tree->AsLclFld();
+                LclVarDsc*     varDsc = lvaGetDesc(lclFld);
                 assert(!lvaInSsa(lclFld->GetLclNum()) || (lclFld->GetFieldSeq() != nullptr));
+
+                // If this is both defined and used here, then assign it unique value if this is
+                // a hidden buffer.
```

Comment:
It would be more truthful to the model to assign the VN at the point of the call.

Also, "location" nodes such as this one should be getting `VNForVoid`.